sympy/integrals/meijerint.py

```diff
@@ -2051,9 +2051,15 @@ def meijerint_inversion(f, x, t):
     if not _is_analytic(f, x):
         _debug('But expression is not analytic.')
         return None
-    # We filter out exponentials here. If we are given an Add this will not
+    # Exponentials correspond to shifts; we filter them out and then
+    # shift the result later.  If we are given an Add this will not
     # work, but the calling code will take care of that.
     shift = 0
+
+    if f.func == exp:
+        # exp would work if written as a Mul
+        f = Mul(2, S.Half, f, evaluate=False)
+
```

Comment:
I would rather try to do something like this:

    if f.is_Mul:
        args = list(f.args)
    elif isinstance(f, exp):
        args = [f]
    else:
        args = None

    if args:
        newargs = []
        ...